plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/editor/ActiveEditorChangeListener.java

```diff
@@ -95,26 +113,39 @@ private void handleEditorChange(final ITextEditor editor) {
         }, DEBOUNCE_DELAY_MS);
     }
 
-    private Map<String, Object> createActiveEditorParams(final ITextEditor editor) {
+    private Map<String, Object> createActiveEditorParams(final Object editor) {
         Map<String, Object> params = new HashMap<>();
         if (editor != null) {
-            Optional<String> fileUri = QEclipseEditorUtils.getOpenFileUri(editor.getEditorInput());
-            if (fileUri.isPresent()) {
-                Map<String, String> textDocument = new HashMap<>();
-                textDocument.put("uri", fileUri.get());
-                params.put("textDocument", textDocument);
-                QEclipseEditorUtils.getSelectionRange(editor).ifPresent(range -> {
-                    Map<String, Object> cursorState = new HashMap<>();
-                    cursorState.put("range", range);
-                    params.put("cursorState", cursorState);
-                });
+            if (editor instanceof ITextEditor textEditor) {
+                Optional<String> fileUri = QEclipseEditorUtils.getOpenFileUri(textEditor.getEditorInput());
+
+                if (fileUri.isPresent()) {
+                    Map<String, String> textDocument = new HashMap<>();
+                    textDocument.put("uri", fileUri.get());
+                    params.put("textDocument", textDocument);
+                    QEclipseEditorUtils.getSelectionRange(textEditor).ifPresent(range -> {
+                        Map<String, Object> cursorState = new HashMap<>();
+                        cursorState.put("range", range);
+                        params.put("cursorState", cursorState);
+                    });
+                }
+            } else if (editor instanceof IEditorPart editorPart) {
+                var input = editorPart.getEditorInput();
+                var file = ((FileEditorInput) input).getFile();
+                var uri = file.getFullPath();
```

Comment:
Why not use  ` QEclipseEditorUtils.getOpenFileUri(part.getEditorInput()) ` here as well? You already have overloaded the `getOpenFilePath` to handle abap paths, so it should return correctly. Might need slight modifications but we should be able to use a central logic in QEclipseEditorUtils to work across the board to handle both general and abap cases